src/ArmTemplates/Creator/TemplateCreators/TagTemplateCreator.cs

```diff
@@ -24,51 +26,64 @@ public TagTemplateCreator(ITemplateBuilder templateBuilder)
 
         public Template CreateTagTemplate(CreatorParameters creatorConfig)
         {
-            // create empty template
-            Template tagTemplate = this.templateBuilder.GenerateEmptyTemplate().Build();
+            var tagTemplate = this.templateBuilder.GenerateTemplateWithApimServiceNameProperty().Build();
+            var tagsDictionary = new Dictionary<string, string>();
 
-            // add parameters
-            tagTemplate.Parameters = new Dictionary<string, TemplateParameterProperties>
+            void AddTagNameToDictionary(string tagName, Dictionary<string, string> tagsDictionary)
```

Comment:
1. what is the reason to make method inside a method? It is a little bit frustrating for me at first sight
2. Even if you want to use inner-method approach, then there is no reason for parameter `Dictionary<string, string> tagsDictionary` - it can reuse the `tagsDictionary` itself

Reply:
We can also build a collection of resources here - just add that `TagDictionary` to `TagResources` and we will do 2 things:
- cache tags dictionary for future, if needed
- build resources for current template built right away